### PR TITLE
Implement Kbet

### LINKED ISSUE #10: Implement kbet
A port to python is necessary, but the core logic of kbet is a chi squared test.

scib implementation:

https://github.com/theislab/scib/blob/main/scib/metrics/kbet.py

kbet paper:

https://www.nature.com/articles/s41592-018-0254-1

### PR DESCRIPTION
**Pull request by adamgayoso**
Reimplement kbet and scib cell type version of kbet

Instead of the diffusion done in scib we use diffusion maps which approximate this via some simple linear algebra; therefore, this implementation is much more efficient.

I got the diffusion map funcs from scanpy. We can later refactor these to use jax sparse (if it ends up working out).

This gives approximately similar results to scib kbet on the lung atlas data, and the accuracy improves as the number of diffusion components increases

Fixes #10 

Our implementation (compare the rejection column in ours to the original):

Screenshot 2022-12-09 at 5 33 38 PM

avg: 0.37
time: 15 seconds

scib:

Screenshot 2022-12-09 at 4 05 24 PM

avg: 0.38
time: 2 min 40 seconds

### CODE REVIEW

File: src/scib_metrics/_lisi.py [hunk 1]
```diff
@@ -44,7 +25,8 @@ def lisi_knn(X: csr_matrix, labels: np.ndarray, perplexity: float = None) -> np.
     lisi
         Array of shape (n_cells,) with the LISI score for each cell.
     """
-    knn_dists, knn_idx = _convert_knn_graph_to_idx(X)
+    labels = np.asarray(pd.Categorical(labels).codes)
```
Comment (martinkim0) [inline src/scib_metrics/_lisi.py:28]: Could this be factored out into utils?

Comment (adamgayoso) [reply src/scib_metrics/_lisi.py:28]: made an issue for that

**Comment by adamgayoso** [comment]
cc @mumichae, tl;dr is that we have kbet in python and it's fast!

File: src/scib_metrics/_kbet.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+from ._types import NdArray
+
+logger = logging.getLogger(__name__)
+
+
+def _chi2_cdf(df: Union[int, NdArray], x: NdArray) -> float:
+    """Chi2 cdf.
+
+    See https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.chdtr.html
+    for explanation of gammaincc.
```
Comment (adamgayoso) [inline src/scib_metrics/_kbet.py:24]:
```suggestion
    for explanation of gammainc.
```

**Pull request merged**